lib/AST/Module.cpp

```diff
@@ -2599,6 +2618,15 @@ bool ModuleDecl::isImportedAsSPI(Identifier spiGroup,
   return importedSPIGroups.count(spiGroup);
 }
 
+bool ModuleDecl::isImportedAsWeakLinked(const Decl *targetDecl) const {
+  const auto *declaringModule = targetDecl->getModuleContext();
+  for (auto file : getFiles()) {
+    if (file->importsModuleAsWeakLinked(declaringModule))
```

Comment:
Should we diagnose some source files use `@ _weakLinked import Foo` but some others use `import Foo`?

Reply:
Yes, I'm planning to do that in an upcoming PR.